### DISCUSSION TITLE
Query the content size and cancel the query in the Progress Monitor

### DISCUSSION THREAD

**Discussion by Skillkiller** [Kong/unirest-java#456] [General]
Hello,
is there any way to find out the file size before the download starts? I only want to download files smaller than 50MB.
Furthermore, is it possible to cancel the request via the ProgressMonitor? I process the requests asynchronously.

**Comment by ryber** [comment]
Yes, if the progress monitor were to throw an exception it will stop the download. You will end up with a HttpResponse with a status of 200 (the servers response), and a parsing error and a null body. Something like this:

```
HttpResponse<File> h = Unirest.get("http://somwhere/spidey.jpg")
                .downloadMonitor((field, fileName, bytesWritten, totalBytes) -> {
                    if (totalBytes > 52428800) {
                        throw new RuntimeException("too big!");
                    }
                })
                .asFileAsync(targeFolder.getPath() + "/spidey.jpg")
                .get();
```

**Reply by ryber** [reply]
not that totalbytes comes from the ```Content-Length``` header

**Comment by Skillkiller** [comment]
Thank you for the quick reply. This is exactly what I needed. I can cancel the requests directly should the content length header be submitted and if not, I can download up to maximum limit.
Thanks a lot.

**Comment by Skillkiller** [comment]
```java
        try {
            Unirest.get("http://speedtest.tele2.net/1GB.zip").downloadMonitor((field, fileName, bytesWritten, totalBytes) -> {
                throw new RuntimeException("Too big!");
            }).asBytesAsync().get();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        } catch (ExecutionException e) {
            throw new RuntimeException(e);
        }
```
The example shown above runs completely through for me without error? I would have assumed that the download aborts. Also by means of sout I get no outputs. It looks to me as if the download monitor is not called?

I use the version 3.13.11

**Reply by ryber** [reply]
The progress monitor only works on File type responses

**Reply by Skillkiller** [reply]
Oh well, with the FileRequest I got it realized as well. Thanks.

**Reply by ryber** [reply]
@Skillkiller I realized last night that there was no real reason for the progress monitor to NOT work on byte[] responses. So I updated 3.13.12 to do that
